src/lib/features/unique-connection/unique-connection-service.test.ts

```diff
@@ -0,0 +1,169 @@
+    eventBus.emit(SDK_CONNECTION_ID_RECEIVED, 'connection2');
+    eventBus.emit(SDK_CONNECTION_ID_RECEIVED, 'connection2');
+    eventBus.emit(SDK_CONNECTION_ID_RECEIVED, 'connection2');
+
+    await uniqueConnectionService.sync();
+
+    const stats = await uniqueConnectionService.getStats();
+    expect(stats).toEqual({ previous: 0, current: 2 });
+});
+
+test('sync first previous bucket', async () => {
+    const eventBus = new EventEmitter();
+    const config = { flagResolver: alwaysOnFlagResolver, getLogger, eventBus };
+    const uniqueConnectionStore = new FakeUniqueConnectionStore();
+    const uniqueConnectionService = new UniqueConnectionService(
+        { uniqueConnectionStore },
+        config,
+    );
+    uniqueConnectionService.listen();
+
+    eventBus.emit(SDK_CONNECTION_ID_RECEIVED, 'connection1');
+    eventBus.emit(SDK_CONNECTION_ID_RECEIVED, 'connection2');
+
+    await uniqueConnectionService.sync();
+
+    eventBus.emit(SDK_CONNECTION_ID_RECEIVED, 'connection3');
+
+    await uniqueConnectionService.sync(() => addHours(new Date(), 1));
+
+    const stats = await uniqueConnectionService.getStats();
+    expect(stats).toEqual({ previous: 3, current: 0 });
+});
+
+test('sync to existing current bucket from the same service', async () => {
+    const eventBus = new EventEmitter();
+    const config = { flagResolver: alwaysOnFlagResolver, getLogger, eventBus };
+    const uniqueConnectionStore = new FakeUniqueConnectionStore();
+    const uniqueConnectionService = new UniqueConnectionService(
+        { uniqueConnectionStore },
+        config,
+    );
+    uniqueConnectionService.listen();
+
+    uniqueConnectionService.count('connection1');
+    uniqueConnectionService.count('connection2');
+
+    await uniqueConnectionService.sync();
+
+    uniqueConnectionService.count('connection1');
+    uniqueConnectionService.count('connection3');
+
+    const stats = await uniqueConnectionService.getStats();
+    expect(stats).toEqual({ previous: 0, current: 3 });
+});
+
+test('sync to existing current bucket from another service', async () => {
+    const eventBus = new EventEmitter();
+    const config = {
+        flagResolver: alwaysOnFlagResolver,
+        getLogger,
+        eventBus: eventBus,
+    };
+    const uniqueConnectionStore = new FakeUniqueConnectionStore();
+    const uniqueConnectionService1 = new UniqueConnectionService(
+        { uniqueConnectionStore },
+        config,
+    );
+    const uniqueConnectionService2 = new UniqueConnectionService(
+        { uniqueConnectionStore },
+        config,
+    );
+
+    uniqueConnectionService1.count('connection1');
+    uniqueConnectionService1.count('connection2');
+    await uniqueConnectionService1.sync();
+
+    uniqueConnectionService2.count('connection1');
+    uniqueConnectionService2.count('connection3');
+    await uniqueConnectionService2.sync();
+
+    const stats1 = await uniqueConnectionService1.getStats();
```

Comment:
is the get stats method implemented to make it easier to test? or is it also used for reporting? I suspect I'll find out later.

Reply:
I was thinking about using it in metrics if we go for the explicit metrics polling for changes mechanism instead of event publishing